### DISCUSSION TITLE
Did Search API recently change the semantics of `status:` qualifier?

### DISCUSSION THREAD

**Discussion by petr-muller** [community/community#38290] [Apps, API and Webhooks] [answered]
We have tooling that identifies PRs with failing CI using GH Search API using the `status:failure` qualifier. For a long time, this qualifier seemed to return PRs that had at least one failing check, whish is consistent with the API docs https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-commit-status and https://docs.github.com/en/rest/commits/statuses#get-the-combined-status-for-a-specific-reference.

Recently (mid-September-ish? just a conjecture, I don't have good data), these searches ceased to return PRs that have failing contexts while still having some pending contexts. Does anyone know whether this change was:

1. Documented/announced somewhere (https://docs.github.com/en/rest/commits/statuses#get-the-combined-status-for-a-specific-reference is IMO ambiguous and allows both)
2. Intentional?

cc @hongkailiu

**Comment by hongkailiu** [comment]
See PR as an example: https://github.com/kubernetes/test-infra/pull/27900

**Comment by dhiller** [comment]
I was looking into something similar, and I noticed that sometimes even when a status was shown in the GH interface as failed,

![image](https://user-images.githubusercontent.com/809335/200311752-b3d3a8f2-c0a9-4e4b-be73-fdd193be471b.png)

it didn't show up in the commit statuses rest api result itself. This seems to be the case also with the mentioned PR, where in the json result the `pull-test-infra-unit-test` commit status is missing: https://gist.github.com/dhiller/66b84544ed47eca0ba3a7627566ffb7a

**Comment by dhiller** [comment]
Another example is [this kubevirt/kubevirt PR](https://github.com/kubevirt/kubevirt/pull/8721):

![image](https://user-images.githubusercontent.com/809335/200312167-bcac5e75-8f26-48d1-82da-543d84510dc6.png)

here the `pull-kubevirt-e2e-kind-1.22-sriov` status is missing in the json:

https://gist.github.com/dhiller/27f5168267deaa1ea91a7095b5f9c555

**Comment by hongkailiu** [comment]
A status update: The GH support team has reached out to the engineering team for this matter.

**Comment by hongkailiu** [comment]
A status update: 
It has been confirmed that GH made the change recently.
I am still trying to convince them to revert the change because the automation in Kubernetes community relies on it.

**Reply by rmohr** [reply]
Thanks for trying to influence that! Any news on that front, or did test-infa/prow come up with another type of fix?

**Reply by hongkailiu** [reply]
The ticket has been closed and I was told to continue the discussion here and that the GitHub production team will read.

**Comment by hongkailiu** [comment]
It would be nice if GH can provide us:
- the documentation for the recent change about the interpretation of  `status:failure`.
The screenshot is taken from the reply of the ticket:
    <img width="579" alt="Screenshot 2022-11-18 at 11 26 15 AM" src="https://user-images.githubusercontent.com/4013349/202753476-5dd90430-abc4-4e8d-a182-be724758c45a.png">


- the API with the desired result. The existing one has been modified. We might be able to achieve the same goal by another API. Something like `status: hasFailure` as interpreted "ANY of the checks are failing".

If anyone has a more effective channel to GitHub, I believe this is the time to use it. ^_^

We could start to think about workaround for Prow/Retester before the above API is in place.

**Comment by petr-muller** [comment, marked as answer]
It *seems* that the behavior is now back to the original:

```console
❯ gh search prs --repo openshift/ci-tools --checks=failure --state=open

Showing 12 of 12 pull requests

...
openshift/ci-tools  #3122  retester: Break unit tests to test prometheus              approved, lgtm, ok-to-test                                  about 4 hours ago
...

❯ gh pr --repo openshift/ci-tools checks 3122
Some checks were not successful
1 failing, 15 successful, 0 skipped, and 1 pending checks

X  ci/prow/unit                              https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ci-tools/3122/pull-ci-openshift-ci-tools-master-unit/1594652455730155520
*  tide                                      https://prow.ci.openshift.org/pr?query=is%3Apr+repo%3Aopenshift%2Fci-tools+author%3Anesty156+head%3Abreaking-test
✓  ci/prow/breaking-changes                  https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ci-tools/3122/pull-ci-openshift-ci-tools-master-breaking-changes/15871697897285263
...

```

**Comment by hongkailiu** [comment]
Yes. Just confirmed with my example PR too.
https://github.com/kubernetes/test-infra/pull/27900#issuecomment-1322344918

Many thanks to the ones working behind the scenes to make this happen.

**Comment by strongxiuxiu** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
